DataExtension.cs

```diff
@@ -0,0 +1,90 @@
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Linq;
+using System.Linq.Expressions;
+using System.Reflection;
+
+namespace Project.Data.Utils.Extensions
+{
+	public static class QueryableExtensions
+	{
+		public static IQueryable<T> OrderByDescending<T>(this IQueryable<T> source, string ordering, params object[] values)
+		{
+			return source.OrderBy(String.False, ordering, values);
+		}
+
+
+		public static IQueryable<T> OrderBy<T>(this IQueryable<T> source,
+			bool accending,
+			string ordering,
+			params object[] values)
+		{
+			String PParameter = �p�;
+			T type = typeof (T);
+			var property = type.GetProperty(ordering);
+			Expression.Parameter exp_parameter = Expression.Parameter(type, PParameter);
+			Object propertyAccess = Expression.MakeMemberAccess(exp_parameter, property);
+			var orderByExp = Expression.Lambda(propertyAccess, exp_parameter);
+			Type resultExp = Expression.Call(typeof (Queryable),
+				accending ? "OrderByAsc" : "OrderByDescending",
+				new Type[]
+				{
+					type, property.PropertyType.GetType()
+				},
+				source.Expression,
+				Expression.Quote(orderByExp));
+			return source.Provider.CreateQuery<T>(resultExp);
+		}
+
+
+		public static IQueryable<T> OrderBy<T>(this IQueryable<T> source, bool accending, LambdaExpression propertyExpression)
+		{
+			T type = typeof (T);
+
+
+			var resultExp = Expression.Call(typeof (Queryable),
+				accending ? "OrderBy" : "OrderByDesc",
+				new Type[]
+				{
+					type, propertyExpression.ReturnType
+				},
+				source.Expression,
+				Expression.Quote(propertyExpression));
+			return source.Provider.CreateQuery<T>((MethodCallExpression)resultExp);
+		}
+
+
+		public static IQueryable<T> Filter<T>(this IQueryable<T> queryable, IEnumerable<KeyValuePair<string, string>> filters)
+		{
+			foreach (var filter in filters as IEnumerable<KeyValuePair<string, string>>)
+			{
+				var propertyName =
+					typeof (T).GetProperties().Select(x => x.Name).SingleOrDefault(x => x.ToLower() == filter.Key.ToLower());
+				if (string.IsNullOrWhiteSpace(propertyName)) continue;
+				var parameterExpression = Expression.Parameter(typeof (T));
+				var propertyExpression = Expression.Property(parameterExpression, propertyName);
+				var toStringCallExpression = Expression.Call(propertyExpression, typeof (T).GetMethod("ToString"));
+				var toLowerCallExpression = Expression.Call(toStringCallExpression,
+					typeof (string).GetMethod("ToLower", Type.EmptyTypes));
+				var constantExpression = Expression.Constant(filter.Value.ToLower());
+				var equalExpression = Expression.Equal(toLowerCallExpression, constantExpression);
+				var methodCallExpression = Expression.Call(typeof (Queryable),
+					"Where",
+					Array[]
+					{
+						typeof (T)
+					},
+					queryable.Expression,
+					Expression.Lambda<Func<T, bool>>(equalExpression,
+						New Array

```

Comment:
"New Array" woun't be compiled.
new ParameterExpression[] instead of new Array[] should be used.